test/unit/javascript/fetchButtonDirectiveTests.js

```diff
@@ -96,6 +96,19 @@ describe('fetchButton', function() {
     it('should have the correct scope when successful with showSummaryStats', function() {
         _prepareScope(1, [1,2], true, {foo: {concepts: ['concept'], valid: true}});
         _clickButton('resolve()', 'resolve({result: {allSamples: 1337, subsets: null}})');
+        expect(element.find('span').text()).toBe('Task complete! Go to the "Run Analysis" tab to continue.');
+        expect(element.isolateScope().running).toBe(false);
+        expect(element.isolateScope().loaded).toBe(true);
+        expect(element.isolateScope().allSamples).toEqual(1337);
+    });
+
+    it('should show `Task complete! Go to the "Preprocess" or "Run Analysis" tab to continue.`', function() {
+        // NOTE overriding parts of `beforeEach` here
+        var html = '<fetch-button concept-map="conceptMap" allowed-cohorts="allowedCohorts" has-preprocess-tab="true"></fetch-button>';
```

Comment:
Hi Sascha,

zwei Fragen:

1.
wenn ich das richtig sehe werden $rootScope.conceptMap und
$rootScope.hasPreprocessTab gar nicht benutzt, denn in _prepareScope
wird der entsprechende Wert von .isolateScope() gesetzt!? Oder sind
isolateScope und $rootScope das gleiche (die scopes immer noch ein
bisschen schleierhaft)?

2.
fuer _prepareScope muss ich einen default setzen, diese Funktionalitaet
in javascript scheint noch zu neu zu sein:
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Default_parameters

Was ist Dir lieber:
1. den call zu _prepareScope ueberall anpassen
2. In _prepareScope scope.hasPreprocessTab immer auf das uebergebene
Argument setzen, dann wird es `undefined` wenn ein Funktionsaufruf
diesen Wert nicht uebergibt (sollte auch klappen)
3. in _prepareScope `if (hasPreprocessTab != undefined)... `

Du bist der Meister von master, Du entscheidest. ;-)

Cheers
  Nils

On 06/09/16 10:37, Sascha Herzinger wrote:
> In test/unit/javascript/fetchButtonDirectiveTests.js
> <https://github.com/transmart/SmartR/pull/49#discussion_r77593422>:
> 
>> @@ -96,6 +96,19 @@ describe('fetchButton', function() {
>>      it('should have the correct scope when successful with showSummaryStats', function() {
>>          _prepareScope(1, [1,2], true, {foo: {concepts: ['concept'], valid: true}});
>>          _clickButton('resolve()', 'resolve({result: {allSamples: 1337, subsets: null}})');
>> +        expect(element.find('span').text()).toBe('Task complete! Go to the "Run Analysis" tab to continue.');
>> +        expect(element.isolateScope().running).toBe(false);
>> +        expect(element.isolateScope().loaded).toBe(true);
>> +        expect(element.isolateScope().allSamples).toEqual(1337);
>> +    });
>> +
>> +    it('should show `Task complete! Go to the "Preprocess" or "Run Analysis" tab to continue.`', function() {
>> +        // NOTE overriding parts of `beforeEach` here
>> +        var html = '<fetch-button concept-map="conceptMap" allowed-cohorts="allowedCohorts" has-preprocess-tab="true"></fetch-button>';
> 
> You don't need to overwrite beforeEach. Just expand the model with a
> variable "hasPreprocessTab", link fetch-button to it, default it to
> false in _prepareScope and set it to true if the respective argument is
> given otherwise.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/transmart/SmartR/pull/49/files/d303a9622b8b56c2f2f1a5e06fe47bb0f2f20966#r77593422>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARNlbNyckiBb9IZUXd48UBXwHm_0zJ8uks5qnSZEgaJpZM4J1BDn>.
> 


-- 
Dr. Nils CHRISTIAN
Developer & Bioinformatician

ITTM S.A.
(Information Technology for Translational Medicine)
9 avenue des Hauts-Fourneaux
L-4362 Esch-sur-Alzette
Luxembourg
http://www.ittm-solutions.com

-----
The information transmitted is intended only for the person or entity to
which it is addressed and may contain confidential and/or privileged
material. Any review, retransmission, dissemination or other use of, or
taking of any action in reliance upon, this information by persons or
entities other than the intended recipient is prohibited. If you
received this in error, please contact the sender and delete the
material from any computer.
-----